Update Rule “conduct-a-test-please-internally-and-then-with-the-client/rule” (#4054)
SSWConsulting/SSW.Rules.Content: rules/conduct-a-test-please-internally-and-then-with-the-client/rule.md

---

**Test, test, test!** Testing is the most critical part of any project. Before the delivery of any release, the application must pass an internal "test please". This helps to prevent clients from becoming disillusioned by receiving a bug-riddled application.

<!--endintro-->

![Figure: Do you want users to have good first impressions?](pic16-TestingDoYouWantThemT.gif)  

I﻿t's important to understand the [different types of tests](/the-different-types-of-test) that you can perform and choose appropriate types for the feature or application you're asked to test.

::: greybox

**Does the "Test Please" principle apply to more than code?**
Yes! A "Test Please", aka peer review, highlights unseen errors, proposes new ideas for consideration or confirms the existing work as the best solution. A peer review can also effect cultural change amongst your development team as developers become more open to critiques of their work and get comfortable with a 'continuous learning' environment. A "Test Please" will also be applied to:
* Brief proposals
* Release plans
* Estimates
* Anything else being sent to a client
* Anything else being sent to an employee of a sensitive nature

> * Anything being sent for public consumption, such as newsletters, print documents and advertisements.

Reviewer: ```suggestion
* Anything being sent for public consumption e.g. newsletters, print documents, and advertisements.
```
